modules/large-deployments/pages/hub-auth.adoc

```diff
@@ -62,12 +62,12 @@ hubSessionKey = client.hub.login(HUB_USERNAME, HUB_PASSWORD)
 serverIds = client.hub.listServerIds(hubSessionKey)
 
 # Authenticate each server with its own credentials
-usernames = [HUB_LOGIN for s in serverIds]
+usernames = [HUB_USERNAME for s in serverIds]
```

Comment:
I don't really know how the hub operates and what this script needs to take care of, but this line looks a bit weird to me.
```python
usernames = [HUB_USERNAME for s in serverIds]
``` 
It assigns a list of the same string (`HUB_USERNAME`) with as many elements of that string as there are `serverIds` to `usernames`. The comment above suggest there are different passwords for each server, but that would require something dynamic (e.g. a function call) that returns the correct username and passwords for each server. 
```python
usernames = [calculate_username_for_server(s) for s in serverIds]
passwords = [calculate_password_for_server(s) for s in serverIds]
``` 
But maybe that's not what is needed. Another option to make this example a bit clearer (to me at least) would be to change the above comment to read something like the following.
```python
# Each server uses the credentials set above, client.hub.attachToServers needs
# them passed as lists with as many elements as there are servers.
```

Reply:
Agreed.
The current script seems tailored to the naive test case scenario; i.e., all machines with the same username and password.
It is mentioned in the point 3. just above this script:
```Credentials for each peripheral server are provided to the attachToServers method. This performs authentication against each server’s XMLRPC API endpoint.
```
Adding your comment to the script would make this clear.